po/zh_CN.po

```diff
@@ -172,67 +172,81 @@ msgid ""
 "a logical subset, simply wrap with which(), and take the which() outside the "
 "loop if possible for efficiency."
 msgstr ""
+"i 为 '%s' 类型。i 必须为整（integer）型，或数值（numeric）型（将被强制转换成整型并发出警告）。如果 i 为"
+"一逻辑向量（logical）的子集，请直接将它传给 which()，且如果可能的话将 which() 放置于"
+"循环之外以保持效率。"
 
 #: assign.c:331
 #, c-format
 msgid "i[%d] is %d which is out of range [1,nrow=%d]."
-msgstr ""
+msgstr "i[%d] 为 %d 且超出了范围 [1,nrow=%d]。"
 
 #: assign.c:334
 #, c-format
 msgid "Assigning to %d row subset of %d rows\n"
-msgstr ""
+msgstr "正在为 %d 行（总数为 %d 行）进行赋值\n"
 
 #: assign.c:342
 #, c-format
 msgid "Added %d new column%s initialized with all-NA\n"
-msgstr ""
+msgstr "添加了 %d 个新列 %s 并全部初始化为 NA \n"
 
 #: assign.c:347
 msgid "length(LHS)==0; no columns to delete or assign RHS to."
-msgstr ""
+msgstr "length(LHS)==0；未删除任何列或将右手侧（RHS）赋值给任何列。"
 
 #: assign.c:361
 msgid ""
 "set() on a data.frame is for changing existing columns, not adding new ones. "
 "Please use a data.table for that. data.table's are over-allocated and don't "
 "shallow copy."
 msgstr ""
+"set() 用于数据框（data.frame）是为了修改现有列，而非添加新列。"
+"请使用 data.table 来完成上述操作。data.table 采用超额分配（over-allocated）并且不"
+"进行浅拷贝（shallow copy）。"
 
 #: assign.c:372
 msgid ""
 "Coerced j from numeric to integer. Please pass integer for efficiency; e.g., "
 "2L rather than 2"
 msgstr ""
+"将 j 从数值（numeric）型强制转换为整（integer）型。为了保持效率请直接传入整型，如"
+"2L 而非 2"
 
 #: assign.c:375
 #, c-format
 msgid ""
 "j is type '%s'. Must be integer, character, or numeric is coerced with "
 "warning."
 msgstr ""
+"j 为 '%s' 型。j 必须为整（integer）型、字符串（character）型、或数值（numeric）型（将被强制转换成整型"
+"并发出警告）。"
 
 #: assign.c:377
 msgid ""
 "Can't assign to the same column twice in the same query (duplicates "
 "detected)."
 msgstr ""
+"在一次查询语句中无法对同一列赋值两次（检测出
```

Comment:
在一次查询中无法对同一列赋值两次（检测出
重复项）。

Reply:
I also considered simply using`查询` not `查询语句`, but thought maybe the latter would provide more context to the user. Could you explain what `query` means here? Does it mean the action to be conducted using `set`? If this is the case, is there any better word to describe this?

Reply:
I think data.table design borrowed a lot from database queries. Thus a `query` is the step happen in `i` which filter the data, then `j` do the calculation (assignment). Since there is no explicit query sentence in the data.table syntax (like a SQL query), I think it's better to use 查询 not 查询语句.

Reply:
Looking at `query` as the step in `i` indeed make more sense. I agree `查询` is a better one here.

Reply:
I need to correct myself. The whole command is a query, some part do the filtering in i, some part do some manipulation in j. This is the same in SQL as you can select some and create new with functions. 查询 is still better than 查询语句 because for a database query we usually don't need to translate it as 查询语句.

Reply:
Thanks for explaining this!

Reply:
@dracodoc is right, we're using it a bit allusively here.

To me, in `data.table`, `query` == any call to `[.data.table`